R/decisiontreeutils.R

```diff
@@ -0,0 +1,363 @@
+  }
+
+  if(rpart_params$cp < 0 || rpart_params$cp > 1) {
+    stop.Alteryx("The complexity parameter must be between 0 and 1. Please try again.")
+  }
+
+  if(is.na(as.numeric(config$the.cp)) && !(config$the.cp == "Auto" || config$the.cp == "")) {
+    stop.Alteryx("The complexity parameter provided is not a number. Please enter a new value and try again.")
+  }
+}
+
+#' Creation of components for model object evaluation
+#'
+#' @param config list of config options
+#' @param data list of datastream inputs
+#' @return list with components needed to create model
+createDTParams <- function(config, data) {
+  # use lists to hold params for rpart and rxDTree functions
+  rpart_params <- rxDTree_params <- list()
+
+  # Determine if a compute context is being used, and whether it is XDF
+  # The mechanism for doing this depends on how the information is transfered
+  xdf_properties <- getXdfProperties("#1")
+  is_XDF <- xdf_properties$is_XDF
+  xdf_path <- xdf_properties$xdf_path
+
+  # get data param
+  the.data <- data$data_stream1
+  data_names <- names(the.data)
+  rpart_params$data <- quote(the.data)
+  rxDTree_params$data <- quote(xdf_path)
+
+  # Get the field names
+  names_list <- getNamesFromOrdered(config$used.weights, data_names)
+  name_y_var <- names_list$y
+  names_x_vars <- names_list$x
+  name_weight_var <- names_list$w
+
+  # use field names to get formula param
+  dtree_formula <- makeFormula(names_x_vars, name_y_var)
+  rpart_params$formula <- rxDTree_params$formula <- dtree_formula
+
+  # get weights param
+  weight_arg <- ifelse(config$used.weights, name_weight_var, NULL)
+  rpart_params$weights <- rxDTree_params$pweights <- weight_arg
+
+  # get method and parms params
+  if(config$select.type) {
+    if(config$classification) {
+      rpart_params$method <- rxDTree_params$method <- "class"
+      if(config$use.gini) {
+        rpart_params$parms <- rxDTree_params$parms <- quote(list(split = "gini"))
+      } else {
+        rpart_params$parms <- rxDTree_params$parms <- quote(list(split = "information"))
+      }
+    } else {
+      rpart_params$method <- rxDTree_params$method <- "anova"
+    }
+  }
+
+  # get surrogate param
+  if(config$use.surrogate.0) {
+    surrogate <- 0
+  } else if(config$use.surrogate.1) {
+    surrogate <- 1
+  } else {
+    surrogate <- 2
+  }
+
+  rpart_params$usesurrogate <- rxDTree_params$useSurrogate <- surrogate
+
+  # get max bins param
+  if(is_XDF && !is.na(as.numeric(config$max.bins))) {
+    max_bins <- config$max.bins
+    if(max_bins < 2) {
+      stop("The minimum bins is 2")
+    } else {
+      rxDTree_params$maxNumBins <- max_bins
+    }
+  }
+
+  # other parameters
+  rpart_params$minsplit <- rxDTree_params$minSplit <- config$min.split
+  rpart_params$minbucket <- rxDTree_params$minBucket <- config$min.bucket
+  rpart_params$xval <- rxDTree_params$xVal <- config$xval.folds
+  rpart_params$maxdepth <- rxDTree_params$maxDepth <- config$max.depth
+
```

Comment:
Why follow so many different conventions here? I see `camelCase`, `dotted` and `nodots`. This can get really confusing very soon.